app/src/main/kotlin/com/wire/android/ui/common/snackbar/SnackBarViewModel.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+
+package com.wire.android.ui.common.snackbar
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.wire.android.util.ui.UIText
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+data class SnackBarState(
+    val showSnackBar: Boolean = false,
+    val message: UIText? = null
+)
+
+@HiltViewModel
+class SnackBarViewModel @Inject constructor(private val showSnackBar: ShowSnackBarUseCase) : ViewModel() {
+
+    private val _snackBarMessage = MutableSharedFlow<SnackBarState>()
+    val snackBarMessage = _snackBarMessage.asSharedFlow()
+
+    init {
+        viewModelScope.launch {
+            showSnackBar.observerSnackBarStatus.collect { result ->
+                result.message?.let {
+                    _snackBarMessage.emit(SnackBarState(message = it))
+                }
+            }
+        }
+    }
```

Comment:
I wonder how this will work with multiple composable elements firing snackbar events 🤔 
Like before we had a snackbar per screen, navigating back would dismiss any not handler request to show a snackbar, with this approach one thing I am concerned how it would work with smaller composables firing events to show a snackbar, if we navigate back those events are still processed, I guess that is not a problem siunce snackbar is a global thing, but perhaps there is more of a risk of a race-condition, for example, user clicks on something on Screen B, there is Composable element that emits a value to show a snackbar, we navigate back from screen B to screen A, the user clicks something that shows the snackbar, what if the event from snackbar B arrived after the one from screen A ? we would be showing a snackbar that should not be shown I am not sure if MutableSharedFlow allows this behaviour because perhaps it does guarantee the order, 

But i think it is worth to at least unit test it a bit, we were having this risk before this approach to :P but the chance of it happening were way less